framework/src/org/checkerframework/framework/type/treeannotator/ImplicitsTreeAnnotator.java

```diff
@@ -83,12 +101,8 @@ public ImplicitsTreeAnnotator(AnnotatedTypeFactory atypeFactory) {
                 continue;
 
             AnnotationMirror theQual = AnnotationUtils.fromClass(atypeFactory.getElementUtils(), qual);
-            for (Class<? extends Tree> treeClass : implicit.treeClasses()) {
-                addTreeClass(treeClass, theQual);
-            }
-
-            for (Kind treeKind : implicit.trees()) {
-                addTreeKind(treeKind, theQual);
+            for(LiteralKind literalKind:implicit.literals()){
+                addLiteralKind(literalKind, theQual);
```

Comment:
Should there be a check whether ALL is used? Does it make sense to have a specific kind and ALL?